python/sgl_jax/srt/layers/moe.py

```diff
@@ -191,44 +191,55 @@ def __init__(
         self.weight_dtype = weight_dtype
         self.dtype = dtype
         self.layer_id = layer_id
-        self.expert_parallel_size = expert_parallel_size
+        self.ep_size = ep_size
         self.mesh = mesh
-        if num_experts % self.expert_parallel_size != 0:
+        if num_experts % self.ep_size != 0:
             raise ValueError(
-                f"num_experts({num_experts}) must be divisible by expert_parallel_size ({self.expert_parallel_size})"
+                f"num_experts({num_experts}) must be divisible by ep_size ({self.ep_size})"
             )
+        world_size = self.mesh.shape.get("data", 1) * mesh.shape.get("tensor", 1)
```

Comment:
Here we can use jax.device_count() to obtain the world_size; there’s no need to pass the mesh for the attention part.